app/features/build_runner/build_runner.rb

```diff
@@ -200,6 +202,19 @@ def setup_build_specific_environment_variables
       env_mapping[:CI_BRANCH] = env_mapping[:GIT_BRANCH]
       # env_mapping[:CI_PULL_REQUEST] = nil # TODO: do we have the PR information here?
 
+      # Now that we have the CI specific ENV variables, let's go through the ENV variables
+      # the user defined in their configuration
+      Services.environment_variable_service.environment_variables.each do |environment_variable|
+        if env_mapping.key?(environment_variable.key.to_sym)
+          # TODO: this is probably large enough of an issue to use the fastlane.ci
+          #       notification system to show an error to the user
+          logger.error("Overwriting CI specific environment variable of key #{environment_variable.key} - " \
+            "this is not recommended")
+        end
+        env_mapping[environment_variable.key.to_sym] = environment_variable.value
+      end
+
+      # Finally, set all the ENV variables for the given build
```

Comment:
Looks like we're not properly cleaning up environment variables if the build crashes (or fails I guess), see https://github.com/fastlane/ci/issues/758 for more information. This is something we have to fix, as we never ever want to leak ENV variables